app/models/manageiq/providers/amazon/agent_coordinator.rb

```diff
@@ -152,43 +140,75 @@ def deploy_agent
       :max_count            => 1,
       :min_count            => 1,
       :placement            => {:availability_zone => zone_name},
-      :security_group_ids   => [security_group_id],
-      :subnet_id            => subnets[0].subnet_id,
-      :tag_specifications   => [{:resource_type => "instance", :tags => [{:key => "Name", :value => label}]}]
+      :tag_specifications   => [{:resource_type => "instance", :tags => [{:key => "Name", :value => label}]}],
+      :network_interfaces   => [{
+        :associate_public_ip_address => true,
+        :delete_on_termination       => true,
+        :device_index                => 0,
+        :subnet_id                   => subnets[0].subnet_id,
+        :groups                      => [security_group_id]
+      }],
     ).first
     ec2.client.wait_until(:instance_status_ok, :instance_ids => [instance.id])
 
     _log.info("Start to load SSA application, this may take a while ...")
 
     setup_agent(instance)
-    _log.info("SSA agent is ready to receive requests.")
+    _log.info("Docker #{docker_image} is loaded. Start to heartbeat.")
 
     instance.id
   end
 
   def setup_agent(instance)
-    # register if needed
-    register(instance)
+    # Somehow instance.public_dns_name is empty, need to reinitialize to get it back
+    ip = ec2.instance(instance.id).public_dns_name || raise("Failed to get agent's public ip!")
+    key_name = instance.key_name
+    auth_key = get_keypair(key_name).try(:auth_key)
+    _log.error("Key [#{key_name}] is missing. Cannot SSH to the agent:#{instance.id}") if auth_key.nil?
+
+    ssh = LinuxAdmin::SSH.new(ip, agent_ami_login_user, auth_key)
 
     # prepare work directory
-    ssh_commands(instance, ["sudo mkdir -p #{WORK_DIR}; sudo chmod go+w #{WORK_DIR}"], agent_ami_login_user)
+    ssh.perform_commands(["sudo mkdir -p #{WORK_DIR}"])
```

Comment:
`perform_commands` accepts an array of commands to perform. I was suggesting that the array have 2 strings instead of a string with a `;` between the commands.